import{_ as t,c as a,b as o,o as i}from"./app-Bn9GZJrW.js";const r={};function s(n,e){return i(),a("div",null,e[0]||(e[0]=[o('<h1 id="import" tabindex="-1"><a class="header-anchor" href="#import"><span>Import</span></a></h1><p><em>Many projects start with imports, and target <a href="/guide/export">Exports</a>, the former are detailed here.</em></p><p>For strategies on migrating whole projects see <a href="/guide/migrate-to-TaxonWorks">Migrate to TaxonWorks</a>. This includes an overview of the many ways that data can be added to TaxonWorks.</p><h2 id="batch-loaders" tabindex="-1"><a class="header-anchor" href="#batch-loaders"><span>Batch loaders</span></a></h2><p>There are various batch importers available within the <a href="/about/glossary#ui">UI</a>. These are polished to differing degrees and have various benefits and limitations. The required format, and often an example spreadsheet, is provided in the UI. All batch loaders are two-step, allowing for (and requiring) a preview of results before inserting them into the database.</p><ul><li>To explore available batch loaders click on a Data card in the Hub. If batch loader(s) are available then then the batch load link will be enabled.</li><li>Batch importers largely target tab-separated text files, though this is not exclusively the case.</li><li>Notable batch loaders are found in the TaxonNames, Otus, and Sources data cards, though others exist.</li><li>Explore various batch loaders (each data card highlighted in yellow has associated batch loaders at this writing).</li></ul><h3 id="try-a-batch-loader" tabindex="-1"><a class="header-anchor" href="#try-a-batch-loader"><span>Try a batch loader</span></a></h3><p>In your test project,</p><ol><li>Go to the data tab</li><li>Select the Otu Data card</li><li>Click “batch load”</li><li>See instructions in the UI for expected / accepted data types and format.</li><li>Create your own file or use this <a href="link">test file</a> Header column = otu_name Blank lines are skipped Tab-delimited format, UTF-8 encoding, Unix line-endings required</li><li>Browse to your file to select it, click preview</li><li>If data looks as expected, browse to select that file again and click create.</li></ol><p>Batch loaders (as of March 2022) include:</p><ul><li>OTUs <a href="https://docs.taxonworks.org/about/glossary.html#otu-operational-taxonomic-unit" target="_blank" rel="noopener noreferrer">operational taxonomic units</a><ul><li>simple batch load</li><li>data attributes</li><li>simple batch file</li><li>OTU with identifier batch load</li></ul></li><li>collecting events <ul><li>gpx (collecting events with georeferences)</li><li>castor</li></ul></li><li>collection objects <ul><li>castor</li><li>buffered strings</li></ul></li><li>descriptors <ul><li>qualitative descriptors</li><li>modify gene descriptor</li></ul></li><li>sequences <ul><li>Genbank</li><li>Genbank batch</li><li>primers</li></ul></li><li>sources <ul><li>BibTeX</li></ul></li><li>taxon names <ul><li>simple</li><li>castor</li></ul></li><li>asserted distributions <ul><li>simple</li></ul></li><li>namespaces <ul><li>simple</li></ul></li><li>sequence relationships <ul><li>primers batch</li></ul></li></ul><h2 id="darwin-core-archive-dwc-a-import" tabindex="-1"><a class="header-anchor" href="#darwin-core-archive-dwc-a-import"><span>Darwin Core Archive (DwC-A) import</span></a></h2><h3 id="checklist-data" tabindex="-1"><a class="header-anchor" href="#checklist-data"><span>Checklist Data</span></a></h3><p>To upload checklist data, this method supports simple and somewhat more complex taxon name lists. Below you will find examples to guide how to create your own datatset and datasets you can use to try in a sandbox.</p><h3 id="preparing-a-checklist" tabindex="-1"><a class="header-anchor" href="#preparing-a-checklist"><span>Preparing a Checklist</span></a></h3><ul><li>Please check the table below for terms (fields) the importer recognizes and whether or not certain fields are required or have dependencies (e. g. formatting, identifiers). This is the <strong>mapping</strong> step.</li><li>Identifiers are required in the following columns for this method to work (<code>taxonID</code>, <code>acceptedNameUsageID</code>, <code>parentNameUsageID</code>). <ul><li>The <code>originalNameUsageID</code> column must be present for the dataset to import. The software will generate the numbers for you for this column if you don&#39;t fill it out (it duplicates the number in the taxonID column).</li></ul></li><li>Running the exact same dataset in twice will not duplicate names in the case where <ul><li>a) the <code>parentNameUsageID</code> <em>is null</em> AND</li><li>b) you use the <code>Settings</code> option to match on existing names where the <code>parentNameUsageID</code> is null.</li><li>IF <code>parentNameUsageID</code> <em>is null</em> and you do not use the <code>Settings</code> option, the names will be entered (again) as children of <code>Root</code> and will say <code>[GENUS Unspecified]</code>. These would need to be cleaned up by hand after import.</li></ul></li><li>Your data in your spreadsheet first goes through a <code>Staging</code> step. You will be able to edit data in each cell at the point, if need be, before you click on <code>Import</code>.</li><li>Each name you want to import must have its own record row in your dataset. For example, if you will be including <strong>higher classification</strong> data, each of those higher taxa must have their own row. If not, your higher classification data for each taxa <strong>will not import.</strong></li><li>Your dataset needs to be in xlsx, comma (csv), or tab-separated (txt, tsv) format.</li><li>For best results for how diacritics are handled (like umlauts or tildas), ensure your data are UTF-8 encoded.</li></ul><div class="hint-container tip"><p class="hint-container-title">Tips</p><ul><li>This method imports only. It does not update (e. g. fix typos) on a re-try or add more data to a given exiting object in the database.</li><li>Case of <code>taxonRank</code> values doesn&#39;t seem to matter.</li><li>IF you want to match on existing names, what matters are [TO BE VERIFIED]: only the <code>sciName</code> + <code>scientificNameAuthorship</code>. (Note a change to higher classification doesn&#39;t seem to matter to making the match).</li></ul></div><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>taxonID</code></td><td>REQUIRED - a unique identifier for the taxa in this record row</td></tr><tr><td><code>parentNameUsageID</code></td><td>REQUIRED - a unique identifier for asserting the correct parent</td></tr><tr><td><code>parentNameUsage</code></td><td></td></tr><tr><td><code>acceptedNameUsageID</code></td><td>REQUIRED - if the name is a valid one, this matches the <code>taxonID</code></td></tr><tr><td><code>scientificName</code></td><td>REQUIRED</td></tr><tr><td><code>kingdom</code></td><td></td></tr><tr><td><code>class</code></td><td></td></tr><tr><td><code>order</code></td><td></td></tr><tr><td><code>family</code></td><td></td></tr><tr><td><code>genus</code></td><td>REQUIRED</td></tr><tr><td><code>subgenus</code></td><td></td></tr><tr><td><code>specificEpithet</code></td><td>REQUIRED</td></tr><tr><td><code>infraspecificEpithet</code></td><td></td></tr><tr><td><code>taxonRank</code></td><td>REQUIRED - Family, Genus, Tribe, Subtribe, Species, etc (not case sensitive)</td></tr><tr><td><code>scientificNameAuthorship</code></td><td>REQUIRED* - Must provide IF you want to match on existing names in the db (and same format)</td></tr><tr><td><code>originalNameUsageID</code></td><td>REQUIRED - Column must be present. IF all cells empty, software will populate them with taxonID at <code>Staging</code> step</td></tr><tr><td><code>nomenclaturalCode</code></td><td>ICZN, ICN - This can be selected in the importer; does not have to be in the spreadsheet</td></tr><tr><td><code>TW:TaxonNameClassification:Latinized:Gender</code></td><td>note maps directly to the TW datamodel; see TW:&lt;data model&gt;:...</td></tr><tr><td><code>TW:TaxonNameClassification:Latinized:PartOfSpeech</code></td><td>note maps directly to the TW datamodel; see TW:&lt;data model&gt;:...</td></tr><tr><td><code>TW:TaxonNameRelationship:incertae\\_sedis\\_in\\_rank</code></td><td>note maps directly to the TW datamodel; see TW:&lt;data model&gt;:...</td></tr><tr><td><code>TW:TaxonNameClassification:Iczn:Fossil</code></td><td>note maps directly to the TW datamodel; see TW:&lt;data model&gt;:...</td></tr></tbody></table><p>| | <strong>need to search codebase to see if these are supported on import</strong> | | taxonomicStatus | valid, incertae sedis, obsolete combination | | originalNameUsage | | cultivarEpithet | | nameAccordingTo | | nomenclaturalStatus | | taxonRemarks | | references |</p><h3 id="the-checklist-importer" tabindex="-1"><a class="header-anchor" href="#the-checklist-importer"><span>The Checklist Importer</span></a></h3><p>What follows are the simplest steps when uploading names into an empty database. It is possible to <strong>match</strong> on existing names in your TW project in the event you are importing children of those names, for example.</p><ol><li>From the <code>Task</code> list select <code>Darwin Core Archive (DwC-A) import</code></li></ol><figure><img src="https://sfg.taxonworks.org/s/0wsyos" alt="" title="shows the task card labeled DwC-A Import"><figcaption class="figure-left">DwC-A Checklist Importer Task</figcaption></figure><ol start="2"><li>In the importer interface, enter a <code>Description</code> for your dataset</li></ol><figure><img src="https://sfg.taxonworks.org/s/fh5j3y" alt="" title="shows layout of drag and drop importer with description, nomencaltural code, and dataset type fields"><figcaption class="figure-left">DwC-A Checklist Importer Screen in TaxonWorks</figcaption></figure><ol start="3"><li><p>Next, select the <code>Dataset type</code>. In this case, <code>Checklist</code></p></li><li><p>Then, select the relevant <code>Nomenclature code</code></p></li><li><p>Once you prepare your dataset, click to upload it by picking or drag and drop the file.</p><ul><li>Depending on the file type (xlsx, csv, txt, tsv) you will need to verify the <strong>separator</strong> (delimiter) for the fields and strings. With xlsx files, the importer figures this out. With csv (comma) and txt (tab) you will get a pop-up asking you to confirm or pick the correct options.</li><li>In either of these delimiter pop-ups, after you pick or verify, click <code>upload</code>.</li></ul><figure><img src="https://sfg.taxonworks.org/s/6dyixc" alt="" title="popup to verify the delimiters used in the csv file to be uploaded"><figcaption class="figure-left">Checklist CSV file delimiter verification</figcaption></figure><figure><img src="https://sfg.taxonworks.org/s/35agr4" alt="" title="popup to verify the delimiters used in the text file to be uploaded"><figcaption class="figure-left">Checklist TXT file delimiter verification</figcaption></figure></li><li><p>The software will <code>Stage</code> your data now (it will take a few seconds or a bit longer depending on the size of the import).</p></li></ol><figure><img src="https://sfg.taxonworks.org/s/srblkw" alt="" title="showing the data ready to import with editable fields before import"><figcaption class="figure-left">The DwC-A Importer <code>Staging</code> step</figcaption></figure><ul><li>Note at this point, you can sort on the columns and replace values in all or any of the cells if necessary (you cannot edit the header rows). <ul><li>Note your original dataset is stored permanently, but not with values you change after <code>Staging</code>.</li></ul></li></ul><ol start="7"><li>Next click <code>Import</code></li></ol><ul><li>Names will import and you can click on <code>Browse</code> for a given row in your dataset to see the data in TW.</li></ul><figure><img src="https://sfg.taxonworks.org/s/25r8ml" alt="" title="shows the have been imported and gives you a button per row to see each imported record"><figcaption class="figure-left">The DwC-A Checklist Importer <code>Browse</code> after upload</figcaption></figure><ol start="8"><li>If you get error messages, rows with errors don&#39;t upload. You can click where it says <code>Error</code> to get the error message.</li></ol><ul><li>For some errors, you can fix them in the spreadsheet and then try to <code>Import</code> that row/s again.</li><li>For example, you might discover an error message <strong>unparsed tail</strong> for a given cell. Sometimes, it might indicate their is an encoding (diacritic) issue or a hidden character. Try retyping the value for that cell and then click to try re-import of that errored row.</li></ul><ol start="9"><li>In the <code>Import</code> pop-up, note you can select <code>Retry errored records</code> where you&#39;ve changed the data in the relevant cells and then click <code>Start import</code>.</li></ol><figure><img src="https://sfg.taxonworks.org/s/z1sc7c" alt="" title="pop up with option to retry import of errored record rows that did not upload"><figcaption class="figure-left">DwC-A Checklist Retry errored rows</figcaption></figure><ol start="10"><li>You can always download your original dataset.</li></ol><h4 id="sample-datasets" tabindex="-1"><a class="header-anchor" href="#sample-datasets"><span>Sample Datasets</span></a></h4><p>We offer five different example datasets (in various file formats) differing in complexity and source (e. g. one of them is from the DwC-A file from a <a href="https://treatment.plazi.org/GgServer/summary/FFA0AE675753FFF9FFC5FFF1FFA9530C" target="_blank" rel="noopener noreferrer">Plazi Treatment Bank Treatment</a>). Please use them to try out the DwC-A Checklist Importer and as models for your own dataset tests and uploads.</p><h4 id="simplest-basic-checklist" tabindex="-1"><a class="header-anchor" href="#simplest-basic-checklist"><span>Simplest Basic Checklist</span></a></h4><p>This dataset inserts a genus and 5 species in that genus. We provide this sample dataset in 3 file formats, csv, txt, xlsx. It was used to upload names into an empty project (no records in the database).</p><ul><li><a href="/examples/sample-checklist-files/basic-simplest-checklist-20250301.csv">Basic sample dataset - CSV</a></li><li><a href="/examples/sample-checklist-files/basic-simplest-checklist-20250301.txt">Basic sample dataset - TXT</a></li><li><a href="/examples/sample-checklist-files/basic-simplest-checklist-20250301.xlsx">Basic sample dataset - XLSX</a></li></ul><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>Import <code>Settings</code> did not seem to matter in this case since we were not trying to macth on any existing names in the database.</p></div><figure><img src="https://sfg.taxonworks.org/s/t5nljb" alt="" title="image showing fields used and sample data for this simplest upload"><figcaption class="figure-left">Simple DwC-A Checklist</figcaption></figure><h4 id="a-published-genus-with-many-new-species" tabindex="-1"><a class="header-anchor" href="#a-published-genus-with-many-new-species"><span>A Published Genus with many new species</span></a></h4><p>In this use case, we take advantage of the Darwin Core Archive formatted <strong>treatment</strong> files that Plazi produces when it pulls names out of existing published literature. With these treatment files you need to add or adjust very few fields (term) headers and the identifiers you need are already in place. This dataset adds 300 names, one new genus and 299 new children of that genus. We did test where the validly published genus was also NOT already in the database. We then also tested how to match on an existing Genus already in the database. See the process below.</p><p>If you are adding new children to an existing genus in the database, then be sure to</p><ul><li>Use the <code>Settings</code> option to match on existing names in the database. Note well that in order to match on existing, the <code>scientificName</code> string and <code>scientificNameAuthorship</code> in the dataset must match the database.</li></ul><p>Here is one simple version (derived from Plazi Treatment Bank taxa.txt from inside the DwC-A file for a given treatment). This file will import 300 names. NOT all fields in this file are imported.</p><ol><li><p>From the original taxa.txt file</p><ul><li>we removed all the synonyms, just leaving new species</li><li>we added a row for the Genus, <em>Galeopsomyia</em>, to match the parent in the TW database</li><li>in the genus row, we put a <code>1</code> for <code>taxonID</code>, <code>acceptedNameUsageID</code>, and <code>originalNameUsageID</code>.</li><li>in the <code>parentNameUsageID</code> column we added a <code>1</code> for all the species</li><li>for the <code>scientificNameAuthorship</code> for the genus row, we made sure to match the Author name as it appears in the database.</li><li>we edited the <strong>combinationAuthor</strong> field to match the paper (there was a parsing error in the Plazi Treatment which has been fixed)</li></ul></li><li><p>Dataset</p></li></ol><ul><li><a href="/examples/sample-checklist-files/plazi-sample-checklist/FFA0AE675753FFF9FFC5FFF1FFA9530C.zip">Original zipped treatment containing multiple files</a></li><li><a href="/examples/sample-checklist-files/plazi-sample-checklist/FFA0AE675753FFF9FFC5FFF1FFA9530C%5Ctaxa.txt">Original taxa.txt file</a></li><li><a href="/examples/sample-checklist-files/plazi-treatment-text-file-modified.txt">Modified taxa.txt file</a></li></ul><ol start="3"><li>So, if you have names to upload, it can pay to check <a href="http://plazi.org/treatmentbank/" target="_blank" rel="noopener noreferrer">Plazi Treatment Bank</a> to see if they have already parsed the names of interest from that published literature.</li></ol><p>To test the entire scenario, have a look at the <a href="/examples/sample-checklist-files/plazi-treatment-text-file-modified.txt">Modified taxa.txt file</a> and try using it to import (into a sandbox account). Columns not recognized by the importer will be ignored.</p><p><strong>Note</strong> there are other usefule files in the Plazi Treatment DwC-A pkg</p><ul><li>The (<a href="/examples/sample-checklist-files/plazi-sample-checklist/FFA0AE675753FFF9FFC5FFF1FFA9530C/references.txt">references.txt</a>) that specifies the page numbers for each new taxon name.</li><li>With some work, we could adjust the importer to add or match on an existing source</li><li>We could imaging, on import, adding a citation for that name inside that source on the specific page.</li><li>With the <a href="/examples/sample-checklist-files/plazi-sample-checklist/FFA0AE675753FFF9FFC5FFF1FFA9530C/multimedia.txt">multimedia.txt</a> data we could link to images (figures) that Plazi processing has deposited in Zenodo as part of creating the treatment.</li><li>Using the <a href="/examples/sample-checklist-files/plazi-sample-checklist/FFA0AE675753FFF9FFC5FFF1FFA9530C/occurrences.txt">occurrences.txt</a> we could pull in data from the materials examined information for each specimen cited in the treatement.</li></ul><p>Meanwhile, you can use <code>Citations by Source</code> to easily add the source page numbers provided in the treatment to each citation record in TW.</p><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>Do check the page numbers that the treament file asserts to ensure the paper was parsed correctly.</p></div><h4 id="bryozoa-names-from-a-website" tabindex="-1"><a class="header-anchor" href="#bryozoa-names-from-a-website"><span>Bryozoa names from a website</span></a></h4><p>In this example set, we started with names we could see on the web (bryozoa.net) for the <a href="http://bryozoa.net/annual/taxa2008.html" target="_blank" rel="noopener noreferrer">year 2008</a>. The following files differ only in file format. Each will import 171 names. Note that to create this file, we had to create the identifier columns for (<code>taxonID</code>, <code>acceptedNameUsageID</code>, <code>parentNameUsageID</code>, and <code>originalNameUsageID</code>). (Some testing suggests that you can leave `originalNameUsageID empty and the upload will work. The column must be present however).</p><ul><li><a href="/examples/sample-checklist-files/bryozoa-2008names-checklist.csv">Bryozoa 2008 - CSV</a></li><li><a href="/examples/sample-checklist-files/bryozoa-2008names-checklist.tsv">Bryozoa 2008 - TSV</a></li><li><a href="/examples/sample-checklist-files/bryozoa-2008names-checklist.xlsx">Bryozoa 2008 - XLSX</a></li></ul><h4 id="more-complex-checklists" tabindex="-1"><a class="header-anchor" href="#more-complex-checklists"><span>More Complex Checklists</span></a></h4><p>Delving into more complex scenarios (synonyms for example) here are some examples for you to look at as you plan your name upload strategy. Note that <a href="https://github.com/SpeciesFileGroup/taxonworks/tree/development/spec/files/import_datasets/checklists" target="_blank" rel="noopener noreferrer">in these datasets</a>, the names existed in a source database. So the identifiers were from their own database. This set of upload test files comes from work done by the developer who wrote the <code>Checklist</code> Importer code.</p><h4 id="source-data-from-checklist-bank" tabindex="-1"><a class="header-anchor" href="#source-data-from-checklist-bank"><span>Source data from Checklist Bank</span></a></h4><p><a href="https://github.com/speciesfilegroup/taxonworks/issues/3658" target="_blank" rel="noopener noreferrer">See recent work</a> to show how you can use / modify datasets from Checklist Bank for importing in to Taxonworks.</p><hr><h3 id="occurrence-data" tabindex="-1"><a class="header-anchor" href="#occurrence-data"><span>Occurrence Data</span></a></h3><p>To upload occurrence data, TW offers the ability to use a DwC Archive file format. <em>For occurrences, the importer is presently limited to vouchered specimen data records.</em></p><p>To use this approach you must have your specimen data in a single spreadsheet-style format that can be export as &quot;CSV&quot;.</p><p>Preparing for an import follows the following general procedures:</p><ul><li><a href="/guide/import#map-your-data">Map your data</a> (provide a column header) for each column of data to be imported</li><li><a href="/guide/import#configure-taxonworks-for-your-dwc-import">Configure TaxonWorks for your DwC import</a> by creating records that will be used during the import process</li></ul><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>As part of your process you may need to go back and forth between mapping and configuring</p></div><h4 id="map-your-data" tabindex="-1"><a class="header-anchor" href="#map-your-data"><span>Map your data</span></a></h4><p>The DwC importer provides flexibility in importing diverse data. These fall in to several types:</p><ol><li>DwC terms</li><li>User customizable data attributes</li><li>User customizable biocuration classes</li><li>TaxonWorks&#39; model specific attributes</li></ol><p>As headers, these will look like this:</p><table><thead><tr><th><code>catalogNumber</code></th><th><code>TW:DataAttribute:CollectionObject:color</code></th><th><code>caste</code></th><th><code>TW:CollectingEvent:verbatim_collectors</code></th></tr></thead><tbody><tr><td><em>A DwC term mapping</em></td><td><em>A user customizable data attribute</em></td><td>A TW biocuration attribute</td><td><em>A TW specific attribute</em></td></tr></tbody></table><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>A first step is to go through your data and figure out which column header type you&#39;ll need. Start by matching to supported DwC terms, then go on from there.</p></div><h4 id="dwc-term-mapping" tabindex="-1"><a class="header-anchor" href="#dwc-term-mapping"><span>DwC term mapping</span></a></h4><p>When going from DwC, a flat format, to TaxonWorks your moving your data from rows to Things. We can group the DwC terms into classes to reflect where they end up in TaxonWorks.</p><h5 id="record-level-class" tabindex="-1"><a class="header-anchor" href="#record-level-class"><span>Record-level class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>type</code></td><td>It is checked that it equals <code>PhysicalObject</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PhysicalObject</code></td></tr><tr><td><code>institutionCode</code></td><td>Selects the repository for the specimen that is registered with an acronym equal to this value</td></tr><tr><td><code>collectionCode</code></td><td>Paired with <code>institutionCode</code> it is used to select the namespace for <code>catalogNumber</code> from a user-defined lookup table in import settings, the value itself is not imported.</td></tr><tr><td><code>basisOfRecord</code></td><td>It is checked that it equals an expected valid value for term, e.g. <code>PreservedSpecimen</code> or <code>FossilSpecimen</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PreservedSpecimen</code>. <em>For compatibility with GBIF datasets, <code>PRESERVED_SPECIMEN</code> is also allowed.</em></td></tr></tbody></table><h5 id="occurrence-class" tabindex="-1"><a class="header-anchor" href="#occurrence-class"><span>Occurrence class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>catalogNumber</code></td><td>The identifier value for Catalog Number local identifier. The namespace is selected from the namespaces lookup table in import settings queried by <code>institutionCode</code>:<code>collectionCode</code> pair. If you require several records to share the same Catalog Number identifier, you may do so by enabling <code>Containerize specimen with existing ones when catalog number already exists</code> import setting or by distinct <code>recordNumber</code> value.</td></tr><tr><td><code>recordNumber</code></td><td>The identifier value for Record Number local identifier. If not empty the record requires to have the short name of the Namespace to use in a TW-specific column named <code>TW:Namespace:RecordNumber</code>. This DwC term enables the re-use of the same <code>catalogNumber</code> of both existing collection objects and records in the dataset, as the importer assigns related specimens to a container to allow sharing the same Catalog Number identifier.</td></tr><tr><td><code>recordedBy</code></td><td>It is imported as-is in verbatim collectors field of the collecting event. Additionally, the value is parsed into people and assigned as collectors of the CE. Previously existing people are not used unless the data origin is the same dataset the record belongs to, otherwise any missing people are created.</td></tr><tr><td><code>individualCount</code></td><td>The total number of entities associated with the specimen record (e.g. this record may be for a &quot;lot&quot; containing 6 objects).</td></tr><tr><td><code>sex</code></td><td>Selects the biocuration class from the &quot;sex&quot; biocuration group to be assigned as biocuration classification for the specimen.</td></tr><tr><td><code>preparations</code></td><td>Selects an existing preparation matching the name with this value.</td></tr></tbody></table><h5 id="event-class" tabindex="-1"><a class="header-anchor" href="#event-class"><span>Event class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>eventID</code></td><td>The identifier for the Collecting Event. If not empty the importer requires a Namespace for it. You may specify a Namespace in a TW-specific column named <code>TW:Namespace:EventID</code> by either using a global identifier type (e.g. <code>Identifier::Global::Uuid</code>, <code>Identifier::Global::Lsid</code>, etc.), or the short name of the Namespace for the Event local identifier. If no namespace is provided, the importer assigns a dataset-specific one with a synthetic name that you can later change. When an existing Collecting Event already has this identifier, the importer re-uses it and the event-related data is ignored.</td></tr><tr><td><code>fieldNumber</code></td><td>The identifier value for Field Number local identifier. If not empty the record requires to have the short name of the Namespace to use in a TW-specific column named <code>TW:Namespace:FieldNumber</code>. The verbatim trip identifier is also populated by this DwC term. When an existing Collecting Event already has this identifier, the importer re-uses it and the event-related data is ignored. IMPORTANT: if a Collecting Event is already matched by <code>eventID</code>, this identifier must exactly match the existing one, otherwise the importer will reject the record. Same rejection will occur if mismatch happens the other way around.</td></tr><tr><td><code>eventDate</code></td><td>The ISO8601-formatted date is split into start year, month and day collecting event fields. If the value is composed of two dates separated by <code>/</code>, then rightmost date is used as end date and split in the same way as start date. If data contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>eventTime</code></td><td>Time is split into time start hour, minute, and second of collecting event</td></tr><tr><td><code>startDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in start year, month, and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>endDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in end year, month and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>year</code></td><td>The start date year of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>month</code></td><td>The start date month of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>day</code></td><td>The start date day of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>verbatimEventDate</code></td><td>Verbatim date of the collecting event</td></tr><tr><td><code>habitat</code></td><td>Verbatim habitat of the collecting event</td></tr><tr><td><code>samplingProtocol</code></td><td>Verbatim method of the collecting event</td></tr><tr><td><code>fieldNotes</code></td><td>Field notes of the collecting event</td></tr></tbody></table><h5 id="location-class" tabindex="-1"><a class="header-anchor" href="#location-class"><span>Location class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>fieldNumber</code></td><td>Verbatim trip identifier of collecting event</td></tr></tbody></table><h5 id="identification-class" tabindex="-1"><a class="header-anchor" href="#identification-class"><span>Identification class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>identifiedBy</code></td><td>A list (concatenated and separated) of names of people, groups, or organizations who assigned the Taxon to the subject. If possible, separate the values in a list with space vertical bar space | (known as a pipe). (e.g. <code>Theodore Pappenfuss | Robert Macey</code>)</td></tr><tr><td><code>dateIdentified</code></td><td>The date on which the subject was determined as representing the Taxon. Best practice is to use a date that conforms to ISO 8601-1:2019 <a href="https://dwc.tdwg.org/terms/#dwc:dateIdentified" target="_blank" rel="noopener noreferrer">see examples</a>.</td></tr></tbody></table><h5 id="taxon-class" tabindex="-1"><a class="header-anchor" href="#taxon-class"><span>Taxon class</span></a></h5><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>nomenclaturalCode</code></td><td>Selects the nomenclatural code for the taxon ranks used when creating protonyms. The value itself is not imported</td></tr><tr><td><code>kingdom</code></td><td>Creates (unless already present) a protonym at kingdom rank</td></tr><tr><td><code>phylum</code></td><td>Creates (unless already present) a protonym at phylum rank</td></tr><tr><td><code>class</code></td><td>Creates (unless already present) a protonym at class rank</td></tr><tr><td><code>order</code></td><td>Creates (unless already present) a protonym at order rank</td></tr><tr><td><code>family</code></td><td>Creates (unless already present) a protonym at family rank</td></tr><tr><td><code>genus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>subgenus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>specificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>infraspecificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificName</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placements</td></tr><tr><td><code>taxonRank</code></td><td>The taxon rank of the most specific protonym</td></tr><tr><td><code>higherClassification</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placement. In case a protonym was not already present, only family-group names will be created, names with classsification higher than family-group not previously registered will result in error. Names at genus rank or lower are ignored and extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificNameAuthorship</code></td><td>Verbatim author of most specific protonym</td></tr></tbody></table><h4 id="taxonworks-mappings" tabindex="-1"><a class="header-anchor" href="#taxonworks-mappings"><span>TaxonWorks mappings</span></a></h4><p>The DwC importer task includes some TW-specific mappings that are neither DwC core terms nor in any DwC extension term lists but instead, direct mappings to predicates in your projects imported as data attributes for collection objects and collecting events, biocuration groups and classes, and as an advanced-use feature you may have direct mappings to model fields.</p><div class="hint-container warning"><p class="hint-container-title">Warning</p><p>If submitting an actual DwC-A zip file and not tab-separated text file or spreadsheet, this TW-specific mappings have to be placed as headers in the core table, and not in meta.xml. If you are replacing a mapping from meta.xml, you must make sure to comment it out and also if inserting colums make sure you do the appropriate adjustments to avoid collision.</p></div><p><em>See <a href="/guide/import#configure-taxonworks-for-your-dwc-import">Configure TaxonWorks for your DwC import</a> for how to create the records referenced in these mappings.</em></p><h5 id="mappings-to-project-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-project-predicates"><span>Mappings to project predicates</span></a></h5><p>In cases where you need to import predicate values targetting the imported collection object or collecting event you may do so by naming the column with a pattern like <code>TW:DataAttribute:&lt;target_class&gt;:&lt;predicate_identifier&gt;</code>. <code>&lt;target_class&gt;</code> may be <code>CollectionObject</code> or <code>CollectingEvent</code>, and the <code>&lt;predicate_identifier&gt;</code> may be the either the name of the predicate or its URI. As an example if you have a predicate registered with name <code>ageInDays</code> and URI <code>http://rs.gbif.org/terms/1.0/ageInDays</code>, both <code>TW:DataAttribute:CollectionObject:ageInDays</code> and <code>TW:DataAttribute:CollectionObject:http://rs.gbif.org/terms/1.0/ageInDays</code> can be used to refer to the same predicate.</p><h5 id="mappings-to-biocuration-groups-and-classes" tabindex="-1"><a class="header-anchor" href="#mappings-to-biocuration-groups-and-classes"><span>Mappings to biocuration groups and classes</span></a></h5><p>The importer is able to map <code>sex</code> into the appropriate biocuration group and select the approriate class according to the value. For additional mappings you may use a special column name pattern to select a biocuration group like <code>TW::BiocurationGroup:&lt;group_identifier&gt;</code> where <code>&lt;group_identifier&gt;</code> can be the name of the biocuration group or its URI. In addition the values must match an existing biocuration class and you may use either its name or URI. For example, if you have a biocuration group registered with name <code>Caste</code> and URI <code>urn:example:ants:caste</code> and biocuration class with name <code>Queen</code> and URI <code>urn:example:ants:caste:queen</code> the following examples do all create the same biocuration classification:</p><table><thead><tr><th>Caste</th><th>urn:example:ants:caste</th></tr></thead><tbody><tr><td>Queen</td><td>urn:example:ants:caste:queen</td></tr><tr><td>urn:example:ants:caste:queen</td><td>Queen</td></tr></tbody></table><h5 id="mappings-to-dwc-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-dwc-predicates"><span>Mappings to DwC predicates</span></a></h5><p>Whenever the importer sees that your project has custom attributes for collecting events and/or collection objects that matches Darwin Core URI terms (<code>http://rs.tdwg.org/dwc/terms/&lt;term&gt;</code>), them will be imported as data attributes regardless of any existing mapping of the same field. This allows to preserve verbatim dataaset value for reference as also to import data from terms not supported by the importer.</p><h5 id="direct-mapping-to-tw-model-fields" tabindex="-1"><a class="header-anchor" href="#direct-mapping-to-tw-model-fields"><span>Direct mapping to TW model fields</span></a></h5><p>This is an advance mapping and requires knowledge of the underlying TW models. The pattern is <code>TW:&lt;model_class&gt;:&lt;field&gt;</code> where model can be either <a href="https://docs.taxonworks.org/develop/Data/models.html#collection-object" target="_blank" rel="noopener noreferrer"><code>CollectionObject</code></a> or <a href="https://docs.taxonworks.org/develop/Data/models.html#collecting-event" target="_blank" rel="noopener noreferrer"><code>CollectingEvent</code></a>, and <code>&lt;field&gt;</code> can be the ones listed below.</p><table><thead><tr><th>Class</th><th>fields</th></tr></thead><tbody><tr><td><code>CollectionObject</code></td><td><code>buffered_collecting_event</code>, <code>buffered_determinations</code>, <code>buffered_other_labels</code>, <code>total</code>,</td></tr><tr><td><code>CollectingEvent</code></td><td><code>document_label</code>, <code>print_label</code>, <code>verbatim_label</code>, <code>field_notes</code>, <code>formation</code>, <code>group</code>, <code>lithology</code>, <code>max_ma</code>, <code>maximum_elevation</code>, <code>member</code>, <code>min_ma</code>, <code>minimum_elevation</code>, <code>elevation_precision</code>, <code>start_date_day</code>, <code>start_date_month</code>, <code>start_date_year</code>, <code>end_date_day</code>, <code>end_date_month</code>, <code>end_date_year</code>, <code>time_end_hour</code>, <code>time_end_minute</code>, <code>time_end_second</code>, <code>time_start_hour</code>, <code>time_start_minute</code>, <code>time_start_second</code>, <code>verbatim_collectors</code>, <code>verbatim_date</code>, <code>verbatim_datum</code>, <code>verbatim_elevation</code>, <code>verbatim_geolocation_uncertainty</code>, <code>verbatim_habitat</code>, <code>verbatim_latitude</code>, <code>verbatim_locality</code>, <code>verbatim_longitude</code>, <code>verbatim_method</code>, <code>verbatim_trip_identifier</code></td></tr></tbody></table><h3 id="configure-taxonworks-for-your-dwc-occurrence-data-import" tabindex="-1"><a class="header-anchor" href="#configure-taxonworks-for-your-dwc-occurrence-data-import"><span>Configure TaxonWorks for your DwC Occurrence data import</span></a></h3><p>To import your DwC you many need to create several types of things in TaxonWorks. These include <a href="Manual/identifiers#namespaces">namespaces</a> and <a href="Manual/customization#controlled-vocabulary-terms">controlled vocabulary terms</a>.</p><h4 id="namespaces" tabindex="-1"><a class="header-anchor" href="#namespaces"><span>Namespaces</span></a></h4><p>In the context of the DwC importer namespaces allow TW to</p><ul><li>Assign an Identifier as a CatalogNumber</li><li>Track uniqueness of each object during the import, helping TW to normalize your data, turning it from rows to Things</li><li>Group your Identifiers (and therefor the CollectionObjects they reference) as coming from a specific place</li></ul><h4 id="controlled-vocabulary-terms" tabindex="-1"><a class="header-anchor" href="#controlled-vocabulary-terms"><span>Controlled vocabulary terms</span></a></h4><p>There are several kinds of CVTs that may be used in the import process.</p><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>All CVTs are created and managed via the <code>Manage controlled vocabulary terms task</code>.</p></div><h5 id="predicates" tabindex="-1"><a class="header-anchor" href="#predicates"><span>Predicates</span></a></h5><p>Think of Predicates as your custom column headers. Predicates are referenced in DataAttributes. Use a Predicate when you want to assign many different values (have rows with many different values) under one heading.</p><h5 id="biocuration-classes" tabindex="-1"><a class="header-anchor" href="#biocuration-classes"><span>Biocuration classes</span></a></h5><p>Think of biocuration classes as custom attributes for your collection objects, things like &#39;male&#39;, &#39;pupa&#39;, or &#39;larva&#39;. These let you assign values useful for your curation of your specimens in a controlled way, ensuring problems like &#39;M.&#39;, &#39;MALE&#39;, &#39;ale&#39; don&#39;t happen in what might otherwise be a &quot;Sex&quot; field. [TODO: reference groups?]. This approach is used when your rows have only a few specific values across the dataset.</p><h3 id="unmapped-columns" tabindex="-1"><a class="header-anchor" href="#unmapped-columns"><span>Unmapped columns</span></a></h3><p>Column headers that can&#39;t be linked via one of the 3 mechanisms are ignored during the import process. This means its important to do some trial runs in a sandbox, or with a smaller dataset to see that your values are mapping over. The <code>Browse collection object</code> task is a good place to check this.</p><div class="hint-container caution"><p class="hint-container-title">Caution</p><p>No warning is given when columns do not map, they are simply ignored.</p></div><div class="hint-container tip"><p class="hint-container-title">Tips</p><p>You can augment your data after import with batch update functionality inside TW. Carefully planning your overal import process can lead to a more efficient overall approach. Sometimes its easier to work in spreadsheets, sometimes within a database.</p></div><h2 id="drag-and-drop" tabindex="-1"><a class="header-anchor" href="#drag-and-drop"><span>Drag and drop</span></a></h2><p>Drag and drop loading of images and documents are accessible in various places in including the <a href="/about/glossary#radial-annotator">Radial annotator</a>, and, notable, <code>Tasks -&gt; New image</code>.</p><h2 id="record-by-record" tabindex="-1"><a class="header-anchor" href="#record-by-record"><span>Record by record</span></a></h2><p>When first learning TaxonWorks, entering records one-at-a-time offers you the opportunity to learn about more of the features in TW and get a feel for how you and others experience the UI.</p><p>For example, you want to enter a specimen record. You have two Tasks enabling you to do this. Choose to use Comprehensive Specimen Digitization Task or the Simple New Specimen Task.</p><h3 id="try-simple-new-specimen" tabindex="-1"><a class="header-anchor" href="#try-simple-new-specimen"><span>Try Simple New Specimen</span></a></h3><p>In your project, try creating a simple new specimen record.</p><ul><li><p>Note you will need to select a namespace. You may find you need to add a namespace before you can do this TW task. Adding a value for namespace ensures your uploaded data records will be unique inside your TW project and across TW projects. In your project, you may also need more than one namespace. [Use Tommy’s INHS Insect Collection as an example, with 12 different namespaces that effectively group the various collections housed at INHS ENT].</p></li><li><p>If you tried the OTU batch loader you can pick one of your OTUs for the name to assign to this specimen.</p></li><li><p>Add an image if you wish</p></li><li><p>Select the Preparation type for this specimen. You may need to add a new value to the dropdown using the New preparation type task.</p></li></ul><h2 id="coming-from-other-software" tabindex="-1"><a class="header-anchor" href="#coming-from-other-software"><span>Coming from other software</span></a></h2><h3 id="scratchpads" tabindex="-1"><a class="header-anchor" href="#scratchpads"><span>Scratchpads</span></a></h3><p>We are in the process of exploring two routes to come from Scratchpads to TaxonWorks.</p><ul><li>The DwC import should work well for occurrence data that is based on collected objects.</li><li>The SFG team is has worked with a select number of individual Scratchpad curators to script the process of transferring their datadata. Contact us if you are interested in what this approach entails. Note that this process takes programming effort that is a limited resource within the SFG.</li></ul>',132)]))}const l=t(r,[["render",s],["__file","import.html.vue"]]),c=JSON.parse('{"path":"/guide/import.html","title":"Import","lang":"en-US","frontmatter":{"sidebarPosition":60},"headers":[{"level":2,"title":"Batch loaders","slug":"batch-loaders","link":"#batch-loaders","children":[{"level":3,"title":"Try a batch loader","slug":"try-a-batch-loader","link":"#try-a-batch-loader","children":[]}]},{"level":2,"title":"Darwin Core Archive (DwC-A) import","slug":"darwin-core-archive-dwc-a-import","link":"#darwin-core-archive-dwc-a-import","children":[{"level":3,"title":"Checklist Data","slug":"checklist-data","link":"#checklist-data","children":[]},{"level":3,"title":"Preparing a Checklist","slug":"preparing-a-checklist","link":"#preparing-a-checklist","children":[]},{"level":3,"title":"The Checklist Importer","slug":"the-checklist-importer","link":"#the-checklist-importer","children":[{"level":4,"title":"Sample Datasets","slug":"sample-datasets","link":"#sample-datasets","children":[]},{"level":4,"title":"Simplest Basic Checklist","slug":"simplest-basic-checklist","link":"#simplest-basic-checklist","children":[]},{"level":4,"title":"A Published Genus with many new species","slug":"a-published-genus-with-many-new-species","link":"#a-published-genus-with-many-new-species","children":[]},{"level":4,"title":"Bryozoa names from a website","slug":"bryozoa-names-from-a-website","link":"#bryozoa-names-from-a-website","children":[]},{"level":4,"title":"More Complex Checklists","slug":"more-complex-checklists","link":"#more-complex-checklists","children":[]},{"level":4,"title":"Source data from Checklist Bank","slug":"source-data-from-checklist-bank","link":"#source-data-from-checklist-bank","children":[]}]},{"level":3,"title":"Occurrence Data","slug":"occurrence-data","link":"#occurrence-data","children":[{"level":4,"title":"Map your data","slug":"map-your-data","link":"#map-your-data","children":[]},{"level":4,"title":"DwC term mapping","slug":"dwc-term-mapping","link":"#dwc-term-mapping","children":[{"level":5,"title":"Record-level class","slug":"record-level-class","link":"#record-level-class","children":[]},{"level":5,"title":"Occurrence class","slug":"occurrence-class","link":"#occurrence-class","children":[]},{"level":5,"title":"Event class","slug":"event-class","link":"#event-class","children":[]},{"level":5,"title":"Location class","slug":"location-class","link":"#location-class","children":[]},{"level":5,"title":"Identification class","slug":"identification-class","link":"#identification-class","children":[]},{"level":5,"title":"Taxon class","slug":"taxon-class","link":"#taxon-class","children":[]}]},{"level":4,"title":"TaxonWorks mappings","slug":"taxonworks-mappings","link":"#taxonworks-mappings","children":[{"level":5,"title":"Mappings to project predicates","slug":"mappings-to-project-predicates","link":"#mappings-to-project-predicates","children":[]},{"level":5,"title":"Mappings to biocuration groups and classes","slug":"mappings-to-biocuration-groups-and-classes","link":"#mappings-to-biocuration-groups-and-classes","children":[]},{"level":5,"title":"Mappings to DwC predicates","slug":"mappings-to-dwc-predicates","link":"#mappings-to-dwc-predicates","children":[]},{"level":5,"title":"Direct mapping to TW model fields","slug":"direct-mapping-to-tw-model-fields","link":"#direct-mapping-to-tw-model-fields","children":[]}]}]},{"level":3,"title":"Configure TaxonWorks for your DwC Occurrence data import","slug":"configure-taxonworks-for-your-dwc-occurrence-data-import","link":"#configure-taxonworks-for-your-dwc-occurrence-data-import","children":[{"level":4,"title":"Namespaces","slug":"namespaces","link":"#namespaces","children":[]},{"level":4,"title":"Controlled vocabulary terms","slug":"controlled-vocabulary-terms","link":"#controlled-vocabulary-terms","children":[{"level":5,"title":"Predicates","slug":"predicates","link":"#predicates","children":[]},{"level":5,"title":"Biocuration classes","slug":"biocuration-classes","link":"#biocuration-classes","children":[]}]}]},{"level":3,"title":"Unmapped columns","slug":"unmapped-columns","link":"#unmapped-columns","children":[]}]},{"level":2,"title":"Drag and drop","slug":"drag-and-drop","link":"#drag-and-drop","children":[]},{"level":2,"title":"Record by record","slug":"record-by-record","link":"#record-by-record","children":[{"level":3,"title":"Try Simple New Specimen","slug":"try-simple-new-specimen","link":"#try-simple-new-specimen","children":[]}]},{"level":2,"title":"Coming from other software","slug":"coming-from-other-software","link":"#coming-from-other-software","children":[{"level":3,"title":"Scratchpads","slug":"scratchpads","link":"#scratchpads","children":[]}]}],"git":{"updatedTime":1747160386000,"contributors":[{"name":"jlpereira","email":"pepelui@outlook.com","commits":1,"url":"https://github.com/jlpereira"},{"name":"Debbie Paul","email":"dlpaul@illinois.edu","commits":13,"url":"https://github.com/Debbie Paul"},{"name":"Hernán Lucas Pereira","email":"ellocodelassembler@gmail.com","commits":2,"url":"https://github.com/Hernán Lucas Pereira"},{"name":"Deborah Paul","email":"dlpaul@illinois.edu","commits":1,"url":"https://github.com/Deborah Paul"},{"name":"mjy","email":"diapriid@gmail.com","commits":10,"url":"https://github.com/mjy"},{"name":"Tommy McElrath","email":"monotomidae@gmail.com","commits":2,"url":"https://github.com/Tommy McElrath"}]},"filePathRelative":"guide/import.md"}');export{l as comp,c as data};
